pkg/distributor/query.go

```diff
@@ -160,7 +161,7 @@ func mergeExemplarSets(a, b []cortexpb.Exemplar) []cortexpb.Exemplar {
 func (d *Distributor) queryIngestersExemplars(ctx context.Context, replicationSet ring.ReplicationSet, req *ingester_client.ExemplarQueryRequest) (*ingester_client.ExemplarQueryResponse, error) {
 	// Fetch exemplars from multiple ingesters in parallel, using the replicationSet
 	// to deal with consistency.
-	results, err := replicationSet.Do(ctx, d.cfg.ExtraQueryDelay, false, func(ctx context.Context, ing *ring.InstanceDesc) (interface{}, error) {
+	results, err := replicationSet.Do(ctx, d.cfg.ExtraQueryDelay, false, false, func(ctx context.Context, ing *ring.InstanceDesc) (interface{}, error) {
```

Comment:
We should also add partial data enabled here

Reply:
Currently `/query_exemplars` endpoint from prometheues [doesn't return any warnings](https://github.com/prometheus/prometheus/blob/main/web/api/v1/api.go#L672), I can try to add it in prometheus side and then add this feature later.